src/libs/Navigation/Navigation.ts

```diff
@@ -410,6 +411,18 @@ function getTopMostCentralPaneRouteFromRootState() {
     return getTopmostCentralPaneRoute(navigationRef.getRootState() as State<RootStackParamList>);
 }
 
+function removeScreenFromNavigationState(screen: Screen) {
+    navigationRef.dispatch((state) => {
+        const routes = state.routes?.filter((item) => item.name !== screen);
+
+        return CommonActions.reset({
+            ...state,
+            routes,
+            index: routes.length < state.routes.length ? state.index - 1 : state.index,
```

Comment:
`index: routes.length - 1` should be enough. The index for the stack points to the last element of the array. 

BTW the current implementation for calculating the index would break if there is more than one screen with given name. It's not possible for the workspace initial but it is possible for other screens. 

Maybe we should limit this function to filter out just the last found route with this name? It may make sense even with fixed `index` implementation to limit this function.

Also maybe we don't have to call reset if we haven't removed any route?